### DISCUSSION TITLE
How to disable field in Edit Form of simple resource

### DISCUSSION THREAD

**Discussion by rajatpatelz** [filamentphp/filament#2739] [Help]
I have created a **simple resource** and want to disable a field only in the edit form.
I have tried the following and it is not working.

```
use Filament\Resources\Pages\Page;
use Filament\Resources\Pages\EditRecord;

->disabled(fn (Page $livewire) => $livewire instanceof EditRecord)
```

**Comment by studiowizjo** [comment]
This works on regular resource, however on simple resource I think it is the same file, so you can try this instead:

```php
->disabled(fn ($record) => !is_null($record))
```

**Reply by danharrin** [reply]
Or, `->disabled(fn ($livewire) => $livewire->mountedTableAction === 'edit')`

**Reply by rajatpatelz** [reply]
Thanks @danharrin

**Reply by danharrin** [reply]
*`->disabledOn('edit')`

**Reply by freddymanullang** [reply]
@danharrin both ->disabled('edit') and ->disablesOn('edit') working for me.
So, what is difference disabled and disabledOn ? When i use disabled and disabledOn ?

**Reply by freddymanullang** [reply]
Never mind @danharrin 
when i using disabled('edit'), the field will disable/readonly for 'create also'. So, if want to disable on edit action only. Must using disabledOn('edit').

**Reply by danharrin** [reply]
thats because `disabled($condition)` will disable if the `$condition` is truthy - 'edit' is truthy

wherewas disabledOn is specifically designed to target certain contexts

**Reply by solstyc** [reply]
Thanks i was looking for a way to do that. I don't think it's in the docs

**Comment by gunawanzai** [comment]
->disabledOn('edit')
that is working for me

**Reply by emineri04** [reply]
why is not working for me :(

**Reply by emineri04** [reply]
oh, i got it, i was trying to use it on a section element instead of a field

**Reply by emineri04** [reply]
is there something to hide a section in edit mode?

**Reply by pboivin** [reply]
->hiddenOn('edit')
